create an enable_telemetry flag for clusters (#1128)
run-house/runhouse: docs/security-and-authentication.rst

    import runhouse as rh
    rh.configs.disable_data_collection()

> Disabling Clusters Observability

Author: Disabling Clusters Observability -> Cluster Observability

> Disabling Cluster Telemetry Recording

Author: Decided to call it Cluster Telemetry Recording after taking a look at some Ray [documentation](https://docs.ray.io/en/latest/_modules/ray/serve/handle.html#). Don't have strong opinion about that though.

---------------------------------------
To disable cluster observability, set the environment variable :code:`enable_observability`
to :code:`False`. Alternatively, set :code:`enable_observability` to :code:`false` in your

> local Runhouse config (:code:`~/.rh/config.yaml`), or in Python:

Reviewer: maybe we make this clear that doing so will disable telemetry across all subsequent clusters? since this is more of a "global" setting

Author: Add more information about what is a cluster observability, before explaining how to disable it.

    import runhouse as rh
    rh.configs.disable_data_collection()

Cluster Observability
---------------------------------------
Runhouse collects various telemetry data by default on clusters. This data will be used to provide better observability
into logs, traces, and metrics associated with clusters. We will not sell data or buy any observability data collected.

> To disable observability for subsequent clusters, set the environment variable :code:`disable_observability`

Reviewer: ```suggestion
To disable observability globally for all clusters, set the environment variable :code:`disable_observability`
```

to :code:`True`. Alternatively, set :code:`disable_observability` to :code:`true` in your
local Runhouse config (:code:`~/.rh/config.yaml`), or in Python:

.. code-block:: python

    import runhouse as rh
    rh.configs.disable_observability()

Please note that the config.yaml and the disable_observability environment variable are **global settings**, which apply to every new cluster you will create using Runhouse.

To disable observability for a specific cluster, you may restart the cluster Runhouse server, using `Runhouse Cluster API <https://www.run.house/docs/api/python/cluster#runhouse.Cluster.restart_server>`__.

.. code-block:: python

    import runhouse as rh
    my_cluster = rh.cluster(name="my-gpu-cluster", instance="CPU:2+").up_if_not()
    my_cluster.restart_server(disable_observability=True)  # for disabling observability on the specific cluster

To enable cluster observability, simply run :code:`my_cluster.restart_server(disable_observability=True)`.

> Alternatively, you may ssh to the cluster, and restart the server using `Runhouse CLI command <https://www.run.house/docs/api/cli#runhouse.main.restart>`__ :

Reviewer: ```suggestion
Alternatively, you may SSH onto the cluster, and restart the Runhouse Server Daemon using the `restart command <https://www.run.house/docs/api/cli#runhouse.main.restart>`__ :
```

    import runhouse as rh
    rh.configs.disable_data_collection()

Cluster Observability
---------------------------------------
Runhouse collects various telemetry data by default on clusters. This data will be used to provide better observability
into logs, traces, and metrics associated with clusters. We will not sell data or buy any observability data collected.

To disable observability globally for all clusters, set the environment variable :code:`disable_observability`
to :code:`True`. Alternatively, set :code:`disable_observability` to :code:`true` in your
local Runhouse config (:code:`~/.rh/config.yaml`), or in Python:

.. code-block:: python

    import runhouse as rh
    rh.configs.disable_observability()

Please note that the config.yaml and the disable_observability environment variable are **global settings**, which apply to every new cluster you will create using Runhouse.

> To disable observability for a specific cluster, you may restart the cluster Runhouse server, using `Runhouse Cluster API <https://www.run.house/docs/api/python/cluster#runhouse.Cluster.restart_server>`__.

Reviewer: a little confusing - could say "restart the Runhouse API server on the cluster"?

Author: how about "restart the Runhouse server on the cluster" or "restart the Runhouse server daemon on the cluster"?

.. code-block:: python

    import runhouse as rh
    my_cluster = rh.cluster(name="my-gpu-cluster", instance="CPU:2+").up_if_not()

>     my_cluster.restart_server(disable_observability=True)  # for disabling observability on the specific cluster

Reviewer: not sure if this is the usage path we want to highlight for disabling observability, bc in this case its entirely possible that the cluster will be started with it enabled

To enable cluster observability, simply run :code:`my_cluster.restart_server(disable_observability=True)`.

> Alternatively, you may ssh onto the cluster, and restart Runhouse Server Daemon using the `restart command <https://www.run.house/docs/api/cli#runhouse.main.restart>`__:

Reviewer: ```suggestion
Alternatively, you may ssh onto the cluster, and restart the Runhouse server daemon using the `restart command <https://www.run.house/docs/api/cli#runhouse.main.restart>`__:
```

.. code-block:: bash

   $ runhouse ssh my-cluster
   $ runhouse restart --disable-observability

To enable cluster observability, simply run :code:`runhouse restart` or :code:`runhouse restart --no-disable-observability`.

> Finally, you may create a brand new Runhouse cluster, and set its :code:`disable_observability` to :code:`True` or :code:`False`, according on your needs.

Reviewer: ```suggestion
Finally, you may create a brand new Runhouse cluster, and set :code:`disable_observability` to :code:`True` if you would like to launch the cluster without any observability.
```

.. code-block:: python

    import runhouse as rh
    my_cluster = rh.cluster(name="my-gpu-cluster", instance="CPU:2+", disable_observability=True).up_if_not()

>     # ...

Reviewer: we can remove these comments
